### PR TITLE
Migrate columnar shuffle configurations to ConfigEntry

### PR DESCRIPTION
**Pull request by pan3793**
# [BUG]/[FEATURE] title

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: docs/configuration/master.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 ---
 license: |
-  Licensed under the Apache License, Version 2.0 (the "License");
-  you may not use this file except in compliance with the License.
-  You may obtain a copy of the License at
+  Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (pan3793) [inline docs/configuration/master.md:3]: updated license header from a generic Apache License to an ASF-specific one.

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -1859,23 +1868,43 @@ object CelebornConf extends Logging {
...
+      .createWithDefault(false)
 
-  def columnarShuffleCompress(conf: CelebornConf): Boolean = {
-    conf.getBoolean("rss.columnar.shuffle.encoding.enabled", defaultValue = false)
-  }
+  val COLUMNAR_SHUFFLE_DICT_ENABLED: ConfigEntry[Boolean] =
+    buildConf("celeborn.columnar.shuffle.dictionary.enabled")
+      .categories("columnar-shuffle")
+      .version("0.2.0")
+      .doc("Whether to use dictionary encoding for columnar-based shuffle data.")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:1883]: I'm not sure about the design purpose of this configuration, the description from my understanding of the code.

**Comment by waitinfuture** [comment]
ping @kerwin-zk

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala [hunk 3]
```diff
@@ -28,7 +28,10 @@ trait RssColumnBuilder {
   /**
    * Initializes with an approximate lower bound on the expected number of elements in this column.
    */
-  def initialize(rowCnt: Int, columnName: String = "", useCompression: Boolean = false): Unit
+  def initialize(
+      rowCnt: Int,
+      columnName: String = "",
+      dictionaryEnabled: Boolean = false): Unit
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala:34]: In order to adapt to new encoding types added later, this suggestion is changed to encodingEnabled

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala [hunk 4]
```diff
@@ -60,7 +63,7 @@ class RssBasicColumnBuilder[JvmType](
   override def initialize(
       rowCnt: Int,
       columnName: String = "",
-      useCompression: Boolean = false): Unit = {
+      dictionaryEnabled: Boolean = false): Unit = {
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala:66]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala [hunk 5]
```diff
@@ -332,7 +335,7 @@ object RssColumnBuilder {
       dataType: DataType,
       rowCnt: Int,
       columnName: String,
-      useCompression: Boolean,
+      dictionaryEnabled: Boolean,
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala:338]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala [hunk 6]
```diff
@@ -362,7 +365,7 @@ object RssColumnBuilder {
         throw new Exception(s"not support type: $other")
     }
 
-    builder.initialize(rowCnt, columnName, useCompression)
+    builder.initialize(rowCnt, columnName, dictionaryEnabled)
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnBuilder.scala:368]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchBuilder.scala [hunk 7]
```diff
@@ -25,7 +25,7 @@ class RssColumnarBatchBuilder(
     schema: StructType,
     batchSize: Int = 0,
     maxDictFactor: Double,
-    useCompression: Boolean = false) extends RssBatchBuilder {
+    dictionaryEnabled: Boolean = false) extends RssBatchBuilder {
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchBuilder.scala:28]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchBuilder.scala [hunk 8]
```diff
@@ -47,7 +47,7 @@ class RssColumnarBatchBuilder(
     if (nativeColumnType == null) {
       null
     } else {
-      if (useCompression && RssDictionaryEncoding.supports(nativeColumnType)) {
+      if (dictionaryEnabled && RssDictionaryEncoding.supports(nativeColumnType)) {
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchBuilder.scala:50]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchBuilder.scala [hunk 9]
```diff
@@ -74,7 +74,7 @@ class RssColumnarBatchBuilder(
         attribute.dataType,
         batchSize,
         attribute.name,
-        useCompression,
+        dictionaryEnabled,
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchBuilder.scala:77]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchSerializer.scala [hunk 10]
```diff
@@ -35,14 +35,14 @@ import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
 class RssColumnarBatchSerializer(
     schema: StructType,
     columnBatchSize: Int,
-    useCompression: Boolean,
+    dictionaryEncodingEnabled: Boolean,
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchSerializer.scala:38]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchSerializer.scala [hunk 11]
```diff
@@ -35,14 +35,14 @@ import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
...
     offHeapColumnVectorEnabled: Boolean,
     dataSize: SQLMetric = null) extends Serializer with Serializable {
   override def newInstance(): SerializerInstance =
     new RssColumnarBatchSerializerInstance(
       schema,
       columnBatchSize,
-      useCompression,
+      dictionaryEncodingEnabled,
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchSerializer.scala:45]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchSerializer.scala [hunk 12]
```diff
@@ -51,7 +51,7 @@ class RssColumnarBatchSerializer(
 private class RssColumnarBatchSerializerInstance(
     schema: StructType,
     columnBatchSize: Int,
-    useCompression: Boolean,
+    dictionaryEncodingEnabled: Boolean,
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssColumnarBatchSerializer.scala:54]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssCompressibleColumnBuilder.scala [hunk 13]
```diff
@@ -38,8 +38,8 @@ trait RssCompressibleColumnBuilder[T <: AtomicType]
   abstract override def initialize(
       rowCnt: Int,
       columnName: String,
-      useCompression: Boolean): Unit = {
-    super.initialize(rowCnt, columnName, useCompression)
+      dictionaryEnabled: Boolean): Unit = {
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssCompressibleColumnBuilder.scala:41]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssCompressibleColumnBuilder.scala [hunk 14]
```diff
@@ -38,8 +38,8 @@ trait RssCompressibleColumnBuilder[T <: AtomicType]
...
+    super.initialize(rowCnt, columnName, dictionaryEnabled)
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssCompressibleColumnBuilder.scala:42]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssNullableColumnBuilder.scala [hunk 15]
```diff
@@ -29,13 +29,13 @@ trait RssNullableColumnBuilder extends RssColumnBuilder {
   abstract override def initialize(
       rowCnt: Int,
       columnName: String,
-      useCompression: Boolean): Unit = {
+      dictionaryEnabled: Boolean): Unit = {
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssNullableColumnBuilder.scala:32]: ditto

File: client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssNullableColumnBuilder.scala [hunk 16]
```diff
@@ -29,13 +29,13 @@ trait RssNullableColumnBuilder extends RssColumnBuilder {
...
 
     nulls = ByteBuffer.allocate(1024)
     nulls.order(ByteOrder.nativeOrder())
     pos = 0
     nullCount = 0
-    super.initialize(rowCnt, columnName, useCompression)
+    super.initialize(rowCnt, columnName, dictionaryEnabled)
```
Comment (kerwin-zk) [inline client-spark/spark-3/src/main/scala/org/apache/spark/sql/execution/columnar/RssNullableColumnBuilder.scala:38]: ditto

**Comment by waitinfuture** [comment]
LGTM except comment from @kerwin-zk

**Comment by pan3793** [comment]
addressed

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**